src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -50,6 +50,15 @@
 _VERSION_CHECK_TIME = 'check_time'
 _VERSION_UPDATE_TIME = 'update_time'
 
+# A list of reserved names that cannot be used as admin username of VM
+DISALLOWED_USER_NAMES = [
+    "administrator", "admin", "user", "user1", "test", "user2",
+    "test1", "user3", "admin1", "1", "123", "a", "actuser", "adm",
+    "admin2", "aspnet", "backup", "console", "guest",
+    "owner", "root", "server", "sql", "support", "support_388945a0",
+    "sys", "test2", "test3", "user4", "user5"
+]
```

Comment:
Is this shared between command modules? If not, putting it in `vm` command module looks more reasonable to me.

Reply:
Yes. It is used in core and vm.

Reply:
should sdk or service team provide the reserved name list?

Reply:
I've got the answer. https://docs.microsoft.com/en-us/rest/api/compute/virtualmachines/createorupdate#osprofile